main.py

```diff
@@ -48,177 +62,352 @@ def validate_yaml_file(yaml_path: Path) -> dict:
-        date_filters = ['all_time', 'month', 'week', '24_hours']
+        date_filters = ["all_time", "month", "week", "24_hours"]
         for date_filter in date_filters:
-            if not isinstance(parameters['date'].get(date_filter), bool):
-                raise ConfigError(f"Date filter '{date_filter}' must be a boolean in config file {config_yaml_path}")
-
-        # Validate positions and locations as lists of strings
-        if not all(isinstance(pos, str) for pos in parameters['positions']):
-            raise ConfigError(f"'positions' must be a list of strings in config file {config_yaml_path}")
-        if not all(isinstance(loc, str) for loc in parameters['locations']):
-            raise ConfigError(f"'locations' must be a list of strings in config file {config_yaml_path}")
+            if not isinstance(parameters["date"].get(date_filter), bool):
+                logger.error(
+                    f"Invalid value for date filter '{date_filter}'. Expected a boolean (True/False)."
+                )
+                raise ConfigError(
+                    f"Date filter '{date_filter}' must be a boolean in config file {config_yaml_path}"
+                )
+
+        # Validate positions and locations
+        if not all(isinstance(pos, str) for pos in parameters["positions"]):
+            logger.error(
+                f"Invalid value in 'positions'. All entries must be strings."
+            )
+            raise ConfigError(
+                f"'positions' must be a list of strings in config file {config_yaml_path}"
+            )
+        if not all(isinstance(loc, str) for loc in parameters["locations"]):
+            logger.error(
+                f"Invalid value in 'locations'. All entries must be strings."
+            )
+            raise ConfigError(
+                f"'locations' must be a list of strings in config file {config_yaml_path}"
+            )
 
         # Validate distance
         approved_distances = {0, 5, 10, 25, 50, 100}
-        if parameters['distance'] not in approved_distances:
-            raise ConfigError(f"Invalid distance value in config file {config_yaml_path}. Must be one of: {approved_distances}")
+        if parameters["distance"] not in approved_distances:
+            logger.error(
+                f"Invalid distance value '{parameters['distance']}'. Must be one of {approved_distances}."
+            )
+            raise ConfigError(
+                f"Invalid distance value in config file {config_yaml_path}. Must be one of: {approved_distances}"
+            )
 
         # Ensure blacklists are lists
-        for blacklist in ['company_blacklist', 'title_blacklist','location_blacklist']:
+        for blacklist in ["company_blacklist", "title_blacklist", "location_blacklist"]:
             if not isinstance(parameters.get(blacklist), list):
-                raise ConfigError(f"'{blacklist}' must be a list in config file {config_yaml_path}")
+                raise ConfigError(
```

Comment:
can we throw config error only once if any of the validation failed? we already have log.error for showing error info

Reply:
done